Mathlib/Data/Vector/MapLemmas.lean

```diff
@@ -56,10 +56,12 @@ theorem map_map (f₁ : β → γ) (f₂ : α → β) :
 end Unary
 
 section Binary
-variable (xs : Vector α n) (ys : Vector β n)
+
+variable {α β : Type} {n : ℕ} (xs : Vector α n) (ys : Vector β n)
+variable {σ₁ σ₂ : Type} {s₁ : σ₁} {s₂ : σ₂}
 
 @[simp]
-theorem mapAccumr₂_mapAccumr_left (f₁ : γ → β → σ₁ → σ₁ × ζ) (f₂ : α → σ₂ → σ₂ × γ) :
+theorem mapAccumr₂_mapAccumr_left {γ ζ : Type} (f₁ : γ → β → σ₁ → σ₁ × ζ) (f₂ : α → σ₂ → σ₂ × γ) :
```

Comment:
Is the missing `*` intentional (also below and above)?
Please make these variables.

Reply:
If you look very closely, you'll see that all lemmas in this section alternatingly have `Type` and `Type*`: the way they are currently written, this is the best one can do. I could extract the Type* case as variables (highlighting the non-polymorphic case); would you prefer this?

Reply:
Does something really break with `Type*` everywhere?

Reply:
I've tried this in [this branch](https://github.com/leanprover-community/mathlib4/compare/MR-vector-typestart): six new errors about "unsolved goals" appear. Be my guest to PR and fix them - I'd like to declare this out of scope of this PR.

Reply:
I have pushed a commit, however, which makes Type* the common case - to emphasize the `Type` is unusual.

Reply:
`Type*` means `Type u` for a new universe variable `u`, where `Type` means `Type 1`: in particular `Type*` is strictly more general than `Type`.